hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkScanSetBuilder.java

```diff
@@ -114,7 +114,9 @@ else if (storedAs.equals("O"))
         }
         if (op != null) {
             ScanOperation sop = (ScanOperation) op;
-            conf.setInt(MRConstants.SPLICE_SPLITS_PER_TABLE, sop.getSplits());
+            int splitsPerTableMin = HConfiguration.getConfiguration().getSplitsPerTableMin();
+            int requestedSplits = sop.getSplits();
+            conf.setInt(MRConstants.SPLICE_SPLITS_PER_TABLE, requestedSplits != 0 ? requestedSplits : splitsPerTableMin);
```

Comment:
Please check that requestedSplits and splitsPerTableMin are positive integers before using them.

Reply:
Thanks @msirek .
Addressed checking the "splice.minimumSplitsPerTable" value be >= 0. It could be 0 if we don't want to set the lower bound.
The "requestedSplit" is checked in FromBaseTable.verifyProperties to make sure it is  > 0. We won't check it again here.

Reply:
@changli6 I didn't mean throw an exception if splitsPerTableMin is negative.  Then the user could cause queries not to run by setting a bad value.  I'm not sure we want to allow them this type of control to cause all spark queries to error out.   I just meant if splitsPerTableMin is negative it should be ignored, for example:

`conf.setInt(MRConstants.SPLICE_SPLITS_PER_TABLE, (requestedSplits != 0) ? requestedSplits : (splitsPerTableMin > 0) ? splitsPerTableMin : 0);`

Reply:
Addressed. Thanks!